dashboard/app/models/foorm/submission.rb

```diff
@@ -11,5 +11,82 @@
 #
 
 class Foorm::Submission < ActiveRecord::Base
-  belongs_to :foorm_form
+  has_one :metadata, class_name: 'Pd::WorkshopSurveyFoormSubmission', foreign_key: :foorm_submission_id
+  belongs_to :form, foreign_key: [:form_name, :form_version], primary_key: [:name, :version]
+
+  SURVEY_CONFIG_KEYS = %w(
+    workshop_course
+    workshop_subject
+    regional_partner_name
+    is_virtual
+    is_friday_institute
+    agenda
+    num_facilitators
+    day
+  )
+
+  # Returns a hash similar to a submission's answer, with the following changes:
+  #   - flattens matrix questions
+  #   - returns readable answer, instead of a key representing a user's answer.
+  def formatted_answers
+    question_answer_pairs = {}
```

Comment:
Would love a better variable name if you have any ideas...or ideas about variable names in general.

Reply:
I also had struggles with naming for parsing forms, there's only so many ways to write question or answer 🤷‍♀️